src/gc/gc.cpp

```diff
@@ -882,12 +882,6 @@ class t_join
                 }
 
                 fire_event (gch->heap_number, time_end, type_join, join_id);
-
-#ifdef JOIN_STATS
-                // parallel execution starts here
-                start[gch->heap_number] = GetCycleCount32();
-                Interlocked::ExchangeAdd(&in_join_total[join_id], (start[gch->heap_number] - end[gch->heap_number])/1000);
-#endif //JOIN_STATS
```

Comment:
Reverse join is different and implementation was incomplete.  Stop collecting statistics instead of collecting bad statistics.